### PR TITLE
LiveList/LiveQuery nullsafety

### LINKED ISSUE #612: Unhandled Exception: Unsupported operation: Cannot add to an unmodifiable list
I am following the ParseLiveListWidget example and getting the following error, I am using the **^3.0.0** version

```
E/flutter (11772): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: Unsupported operation: Cannot add to an unmodifiable list
E/flutter (11772): #0      UnmodifiableListMixin.addAll (dart:_internal/list.dart:129:5)
E/flutter (11772): #1      ParseLiveList._runQuery (package:parse_server_sdk/src/utils/parse_live_list.dart:138:14)
E/flutter (11772): #2      ParseLiveList._init (package:parse_server_sdk/src/utils/parse_live_list.dart:154:47)
E/flutter (11772): #3      ParseLiveList.create (package:parse_server_sdk/src/utils/parse_live_list.dart:28:26)
E/flutter (11772): #4      new _ParseLiveListWidgetState (package:parse_server_sdk_flutter/src/utils/parse_live_list.dart:89:23)
E/flutter (11772): #5      ParseLiveListWidget.createState (package:parse_server_sdk_flutter/src/utils/parse_live_list.dart:51:49)
E/flutter (11772): #6      new StatefulElement (package:flutter/src/widgets/framework.dart:4584:24)
E/flutter (11772): #7      StatefulWidget.createElement (package:flutter/src/widgets/framework.dart:916:38)
E/flutter (11772): #8      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3536:40)
E/flutter (11772): #9      Element.updateChild (package:flutter/src/widgets/framework.dart:3306:18)
E/flutter (11772): #10     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4520:16)
E/flutter (11772): #11     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4667:11)
E/flutter (11772): #12     Element.rebuild (package:flutter/src/widgets/framework.dart:4189:5)
E/flutter (11772): #13     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4474:5)
E/flutter (11772): #14     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4658:11)
E/flutter (11772): #15     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4469:5)
E/flutter (11772): #16     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3541:14)
E/flutter (11772): #17     Element.updateChild (package:flutter/src/widgets/framework.dart:3306:18)
E/flutter (11772): #18     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4520:16)
E/flutter (11772): #19     Element.rebuild (package:flutter/src/widgets/framework.dart:4189:5)
E/flutter (11772): #20     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4474:5)
E/flutter (11772): #21     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4469:5)
E/flutter (11772): #22     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3541:14)
E/flutter (11772): #23     Element.updateChild (package:flutter/src/widgets/framework.dart:3306:18)
E/flutter (11772): #24     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4520:16)
E/flutter (11772): #25     Element.rebuild (package:flutter/src/widgets/framework.dart:4189:5)
E/flutter (11772): #26     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4474:5)
E/flutter (11772): #27     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4469:5)
E/flutter (11772): #28     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3541:14)
E/flutter (11772): #29     Element.updateChild (package:flutter/src/widgets/framework.dart:3306:18)
E/flutter (11772): #30     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4520:16)
E/flutter (11772): #31     Element.rebuild (package:flutter/src/widgets/framework.dart:4189:5)
E/flutter (11772): #32     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4474:5)
E/flutter (11772): #33     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4469:5)
E/flutter (11772): #34     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3541:14)
E/flutter (11772): #35     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6094:32)
E/flutter (11772): #36     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3541:14)
E/flutter (11772): #37     Element.updateChild (package:flutter/src/widgets/framework.dart:3306:18)
E/flutter (11772): #38     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4520:16)
E/flutter (11772): #39     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4667:11)
E/flutter (11772): #40     Element.rebuild (package:flutter/src/widgets/framework.dart:4189:5)
E/flutter (11772): #41     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4474:5)
E/flutter (11772): #42     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4658:11)
E/flutter (11772): #43     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4469:5)
E/flutter (11772): #44     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3541:14)
E/flutter (11772): #45     Element.updateChild (package:fl
```

### PR DESCRIPTION
**Pull request by fischerscode**
**TLDR:** No new features, just fixes arround the ParseLiveList.

While migrating my app, I've noticed LiveList has been broken due to the null-safety migration. I've decided to go through the entire LiveQuery and LiveList part in order to remove those not-null casts (`something!`).

Additionally I've [changed](https://github.com/parse-community/Parse-SDK-Flutter/commit/d5223d13c81639d2b09b39618288ec4ee4f91a94) `object` in `QueryBuilder` from nullable to not-nullable. In theory this is a breaking change, as we formally allowed null as the argument in the default constructor before. But as setting it to null would result in a crash, I think we should be fine with updating the version number to just "3.0.1".

Close #612

### CODE REVIEW

**Comment by RodrigoSMarques** [comment]
Hey @fischerscode.

Very good. Change the version to 3.1.0.

I'm validating other PR and will ask @phillwiggins to release this fix and other improvements next week.

**Comment by fischerscode** [comment]
@RodrigoSMarques
Great!
I've updated the version and changed the test to always use the dart package in "../dart".

**Comment by mtrezza** [comment]
@fischerscode I notice that you have currently access to this repository, granted on an individual basis. Since we are moving to team-managed permissions, I will go ahead and convert your permissions by inviting you to the repository as team member and removing your individually granted access. Please let me know if you have any questions.

**Comment by fischerscode** [comment]
@mtrezza 
Thanks for the invite, I've accepted it and currently there are no questions.
